Question 1 (What's the version of dlt that you installed?)

In [ ]:
#after installing(Use this command) pip install -q "dlt[qdrant]" "qdrant-client[fastembed]"
%pip show dlt
%pip show qdrant-client

Name: dlt
Version: 1.12.3
Summary: dlt is an open-source python-first scalable data loading library that does not require any backend to run.
Home-page: 
Author: 
Author-email: "dltHub Inc." <services@dlthub.com>
License: 
Location: c:\Users\ilias\AppData\Local\Programs\Python\Python312\Lib\site-packages
Requires: click, fsspec, gitpython, giturlparse, hexbytes, humanize, jsonpath-ng, orjson, packaging, pathvalidate, pendulum, pluggy, pytz, pywin32, pyyaml, requests, requirements-parser, rich-argparse, semver, setuptools, simplejson, sqlglot, tenacity, tomlkit, typing-extensions, tzdata, win-precise-time
Required-by: cognee
Note: you may need to restart the kernel to use updated packages.
Name: qdrant-client
Version: 1.14.3
Summary: Client library for the Qdrant vector search engine
Home-page: 
Author: Andrey Vasnetsov
Author-email: andrey@qdrant.tech
License: Apache-2.0
Location: c:\Users\ilias\AppData\Local\Programs\Python\Python312\Lib\site-packages
Requires: grpcio, httpx, numpy, p

Answer is Version 1.12.3

In [5]:
import requests
import dlt

@dlt.resource
def zoomcamp_data():
    docs_url = 'https://github.com/alexeygrigorev/llm-rag-workshop/raw/main/notebooks/documents.json'
    docs_response = requests.get(docs_url)
    documents_raw = docs_response.json()

    for course in documents_raw:
        course_name = course['course']

        for doc in course['documents']:
            doc['course'] = course_name
            yield doc


Question 2 (How many rows were inserted into the zoomcamp_data collection?)

We need to define a destination for that. Let's use the qdrant one:

In [6]:
from dlt.destinations import qdrant

qdrant_destination = qdrant(
  qd_path="db.qdrant", 
)

In [7]:
pipeline = dlt.pipeline(
    pipeline_name="zoomcamp_pipeline",
    destination=qdrant_destination,
    dataset_name="zoomcamp_tagged_data"

)
load_info = pipeline.run(zoomcamp_data())
print(pipeline.last_trace)

c:\Users\ilias\AppData\Local\Programs\Python\Python312\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
Fetching 5 files:   0%|          | 0/5 [00:00<?, ?it/s]
c:\Users\ilias\AppData\Local\Programs\Python\Python312\Lib\site-packages\huggingface_hub\file_download.py:143: UserWarning: `huggingface_hub` cache-system uses symlinks by default to efficiently store duplicated files but your machine does not support them in C:\Users\ilias\AppData\Local\Temp\fastembed_cache\models--Qdrant--bge-small-en. Caching files will still work but in a degraded version that might require more space on your disk. This warning can be disabled by setting the `HF_HUB_DISABLE_SYMLINKS_WARNING` environment variable. For more details, see https://huggingface.co/docs/huggingface_hub/how-to-cache#limitations.
To support symlinks on Windows, yo

Run started at 2025-07-07 18:38:43.283006+00:00 and COMPLETED in 24.43 seconds with 4 steps.
Step extract COMPLETED in 0.80 seconds.

Load package 1751913536.6524866 is EXTRACTED and NOT YET LOADED to the destination and contains no failed jobs

Step normalize COMPLETED in 0.07 seconds.
Normalized data for the following tables:
- zoomcamp_data: 948 row(s)
- _dlt_pipeline_state: 1 row(s)

Load package 1751913536.6524866 is NORMALIZED and NOT YET LOADED to the destination and contains no failed jobs

Step load COMPLETED in 10.20 seconds.
Pipeline zoomcamp_pipeline load step completed in 10.19 seconds
1 load package(s) were loaded to destination qdrant and into dataset zoomcamp_tagged_data
The qdrant destination used c:\Users\ilias\Documents\GitHub\llm-zoomcamp\03-Workshop\db.qdrant location to store data
Load package 1751913536.6524866 is LOADED and contains no failed jobs

Step run COMPLETED in 24.43 seconds.
Pipeline zoomcamp_pipeline load step completed in 10.19 seconds
1 load package

948 rows where inserted to the zoomcamp_data collection

Question 3. Embeddings (When inserting the data, an embedding model was used. Which one?)

In [11]:
import json

meta_path = r"C:\Users\ilias\Documents\GitHub\llm-zoomcamp\03-Workshop\db.qdrant\meta.json"

with open(meta_path, "r") as f:
    meta = json.load(f)

print(meta)


{'collections': {'zoomcamp_tagged_data': {'vectors': {'fast-bge-small-en': {'size': 384, 'distance': 'Cosine', 'hnsw_config': None, 'quantization_config': None, 'on_disk': None, 'datatype': None, 'multivector_config': None}}, 'shard_number': None, 'sharding_method': None, 'replication_factor': None, 'write_consistency_factor': None, 'on_disk_payload': None, 'hnsw_config': None, 'wal_config': None, 'optimizers_config': None, 'init_from': None, 'quantization_config': None, 'sparse_vectors': None, 'strict_mode_config': None}, 'zoomcamp_tagged_data__dlt_pipeline_state': {'vectors': {'fast-bge-small-en': {'size': 384, 'distance': 'Cosine', 'hnsw_config': None, 'quantization_config': None, 'on_disk': None, 'datatype': None, 'multivector_config': None}}, 'shard_number': None, 'sharding_method': None, 'replication_factor': None, 'write_consistency_factor': None, 'on_disk_payload': None, 'hnsw_config': None, 'wal_config': None, 'optimizers_config': None, 'init_from': None, 'quantization_config'

Answer is fast-bge-small-en